### PR TITLE
[MI-2906]: Fix issues in the input component after downgrading bootstrap and react-bootstrap.

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**

### CODE REVIEW

File: src/components/Input/Input.d.ts [hunk 1]
```diff
@@ -94,10 +94,10 @@ export interface InputProps {
     /**
 	 * Handle trigger on clicking the close icon button
 	 */
-    onClose?: React.MouseEventHandler<HTMLElement>;
+    onClose?: React.MouseEventHandler<Button>;
```
[the remark this answers is not in the document]
Comment (SaurabhSharma-884) [reply src/components/Input/Input.d.ts:97]: Added

File: src/components/Input/Input.component.tsx [hunk 2]
```diff
@@ -43,7 +44,8 @@ const DisplayFieldSet = ({value, error, label}: InputProps) => (
  * <Input label='label' iconName='Globe'/>
  * ```
  */
-export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
+// eslint-disable-next-line react/display-name
```
[the remark this answers is not in the document]
Comment (SaurabhSharma-884) [reply src/components/Input/Input.component.tsx:47]: Removed

File: src/components/Input/Input.component.tsx [hunk 3]
```diff
@@ -82,16 +83,20 @@ export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
...
             />
             {searchQuery && (
-                <StyledIconButton onClick={onClose}>
+                <StyledIconButton
+                    onClick={onClose}
+                    bsStyle='primary'
+                >
                     <Icon
                         name='Close'
                         size={12}
```
Comment (aushwin) [inline src/components/Input/Input.component.tsx:102]: use constants for the icon color

File: src/components/Input/Input.component.tsx [hunk 4]
```diff
@@ -43,7 +44,7 @@ const DisplayFieldSet = ({value, error, label}: InputProps) => (
  * <Input label='label' iconName='Globe'/>
  * ```
  */
-export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
+export const Input = forwardRef<FormControl, InputProps>((props, ref) => {
```
Comment (aushwin) [inline src/components/Input/Input.component.tsx:47]: Please destrcutre here

**Comment by aushwin** [comment]
Please fix this error too 
image

File: src/components/Input/Input.component.tsx (hunk 4 above)
Comment (aushwin) [reply src/components/Input/Input.component.tsx:47]: Is this resolved ?

**Review by aushwin — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**